server/tso/global_allocator.go

```diff
@@ -118,63 +151,99 @@ func (gta *GlobalTSOAllocator) GenerateTSO(count uint32) (pdpb.Timestamp, error)
 	}
 
 	// Have dc-locations configured in the cluster, use the Global TSO generation way.
-	// Send maxTS to all Local TSO Allocator leaders to prewrite
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
+	skipCheck := false
 	maxTSO := &pdpb.Timestamp{}
-	// 1. Collect the MaxTS with all Local TSO Allocator leaders first
-	if err := gta.SyncMaxTS(ctx, dcLocationMap, maxTSO); err != nil {
+ESTIMATING_PHASE:
+	// 1. Estimate a MaxTS among all Local TSO Allocator leaders according to the RTT.
+	estimatedMaxTSO, err := gta.estimateMaxTS(count)
+	if err != nil {
 		return pdpb.Timestamp{}, err
 	}
-	// 2. Add the count and make sure its logical part won't overflow after being differentiated.
+SETTING_PHASE:
+	// 2. Precheck to make sure the logical part won't overflow after being differentiated.
 	suffixBits := gta.allocatorManager.GetSuffixBits()
-	gta.preprocessLogical(maxTSO, count, suffixBits)
-	// 3. Sync the MaxTS with all Local TSO Allocator leaders then
-	if err := gta.SyncMaxTS(ctx, dcLocationMap, maxTSO); err != nil {
+	// If precheckLogical returns false, it means the logical part is overflow,
+	// we need to wait a updatePhysicalInterval and retry the estimation.
+	if !gta.precheckLogical(estimatedMaxTSO, suffixBits) {
+		time.Sleep(gta.timestampOracle.updatePhysicalInterval)
+		goto ESTIMATING_PHASE
+	}
```

Comment:
We can move this check into `estimateMaxTS`, thus `goto ESTIMATING_PHASE` can be removed.